tests/test_entity.py

```diff
@@ -1,121 +1,259 @@
+    ):
+        me.Entity("SpontaneousMagnetization", value=1 * u.T, unit="A/m")
 
 
-def test_tuple_as_value():
-    val = (42, 42, 42)
-    e = me.Ms(val)
-    assert np.allclose(e.value, np.array(val))
+# %% check attributes
 
 
-def test_entity_drop_ontology_numpy(onto_class_list):
-    for label in onto_class_list:
-        e = me.Entity(label, 42)
-        root_e = np.sqrt(e)
-        with pytest.raises(AttributeError):
-            _ = root_e.ontology
+def test_attrs_H():
+    e = me.Entity("ExternalMagneticField")
+    assert hasattr(e, "ontology_label")
+    assert hasattr(e, "ontology_label_with_iri")
+    assert hasattr(e, "ontology")
+    assert hasattr(e, "quantity")
+    assert hasattr(e, "value")
+    assert hasattr(e, "unit")
+    assert hasattr(e, "axis_label")
 
 
-def test_entity_drop_ontology_multiply(onto_class_list):
-    for label in onto_class_list:
-        e = me.Entity(label, 42)
-        mul_e = e * e
-        with pytest.raises(AttributeError):
-            _ = mul_e.ontology
+# %% Check repr, str
 
 
-def test_all_labels_ontology(onto_class_list):
-    for label in onto_class_list:
-        _ = me.Entity(label, 42)
+def test_repr_H():
+    a = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
+    e = me.Entity("ExternalMagneticField", value=a)
+    assert e.__repr__() == (
+        "Entity(ontology_label='ExternalMagneticField', "
+        + f"value={np.array(a, dtype=float)!r}, unit='A / m')"
+    )
+    assert eval(repr(e)) == e
 
 
-def test_quantity_as_value():
-    val = 1 * u.A / u.m
-    e = me.Ms(val)
-    assert u.allclose(e.quantity, val)
+def test_repr_Tc():
+    e = me.Entity("CurieTemperature")
+    assert (
+        e.__repr__()
+        == "Entity(ontology_label='CurieTemperature', value=np.float64(0.0), unit='K')"
+    )
+    assert eval(repr(e)) == e
 
 
-def test_wrong_quantity_as_value():
-    val = 1 * u.T
-    with pytest.raises(u.UnitConversionError):
-        me.Ms(val)
+def test_repr_unitless():
+    e = me.Entity("DemagnetizingFactor")
+    assert (
+        e.__repr__()
+        == "Entity(ontology_label='DemagnetizingFactor', value=np.float64(0.0))"
+    )
+    assert eval(repr(e)) == e
 
 
-def test_quantity_as_value_and_unit():
-    val = 1 * u.A / u.m
-    e = me.Ms(val, "A/m")
-    assert u.allclose(e.quantity, val)
+def test_axis_label_H():
+    e = me.Entity("ExternalMagneticField")
+    assert e.axis_label == "External Magnetic Field (A / m)"
 
 
-def test_wrong_quantity_as_value_and_unit():
-    val = 1 * u.T
-    with pytest.raises(u.UnitConversionError):
-        me.Ms(val, "A/m")
+def test_axis_label_Ms():
+    e = me.Entity("SpontaneousMagnetization")
+    assert e.axis_label == "Spontaneous Magnetization (A / m)"
 
 
-def test_wrong_quantity_as_value_and_wrong_unit():
-    val = 1 * u.T
-    with pytest.raises(u.UnitConversionError):
-        me.Ms(val, "T")
+def test_axis_label_Tc():
+    e = me.Entity("CurieTemperature")
+    assert e.axis_label == "Curie Temperature (K)"
```

Comment:
We could probably drop both these test. The one above covers the same case. If the ontology has elements with a single capital letter, it would be good to check one of those (to ensure the regex conversion is fine also when there is nothing to be inserted).